geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert2/validators/SimpleFeatureValidator.scala

```diff
@@ -0,0 +1,267 @@
+/***********************************************************************
+ * Copyright (c) 2013-2019 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.convert2.validators
+
+import java.util.{Date, Locale}
+
+import com.typesafe.scalalogging.LazyLogging
+import org.locationtech.geomesa.curve.BinnedTime
+import org.locationtech.geomesa.utils.classpath.ServiceLoader
+import org.locationtech.geomesa.utils.conf.GeoMesaSystemProperties.SystemProperty
+import org.locationtech.geomesa.utils.geotools.wholeWorldEnvelope
+import org.locationtech.jts.geom.Geometry
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+trait SimpleFeatureValidator {
+
+  /**
+    * Validate a feature
+    *
+    * @param sf simple feature
+    * @return validation error message, or null if valid
+    */
+  def validate(sf: SimpleFeature): String
+}
+
+object SimpleFeatureValidator extends LazyLogging {
+
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+  private lazy val factories = ServiceLoader.load[SimpleFeatureValidatorFactory]()
+
+  // noinspection ScalaDeprecation
+  private lazy val factoriesV1 =
+    ServiceLoader.load[org.locationtech.geomesa.convert.SimpleFeatureValidator.ValidatorFactory]()
+
+  val DefaultValidators = SystemProperty("geomesa.converter.validators", IndexValidatorFactory.Name)
+
+  /**
+    * Default validator names
+    *
+    * @return
+    */
+  def default: Seq[String] = DefaultValidators.get.split(",")
```

Comment:
How often does this get called?  Should this be something that isn't recomputed?

Reply:
I think it's only called when using type inference to create a converter conf, so should not be frequently. I made it a def so that it could be modified via the thread-local value.

Reply:
Hmm...  from reading the code, I kinda expect DefaultValidators to be static and unchangeable. 

Not a problem; just my expectation:).